client/ayon_core/pipeline/create/product_name.py

```diff
@@ -183,7 +182,10 @@ def get_product_name(
             fill_pairs[key] = value
 
     try:
-        return template.format(**prepare_template_data(fill_pairs))
+        return StringTemplate.format_template(
```

Comment:
```suggestion
        return StringTemplate.format_strict_template(
```

Reply:
It should also probably be explicitly converted to `str` because this does return `TemplateResult` which might cause issues in some DCCs.